### PR TITLE
Updating to use new PE api

### PR DESCRIPTION
**Pull request by aaronjeline**
*Issue #, if available:*
#271 
*Description of changes:*

Recent PR changed the partial eval API, this changes the test target to keep up.

### CODE REVIEW

File: cedar-drt/fuzz/fuzz_targets/partial-eval.rs [hunk 1]
```diff
@@ -146,32 +204,32 @@ fuzz_target!(|input: FuzzTargetInput| {
...
+        } else {
+            // If we are not able to reach a concrete decision:
+            // First we ensure that if we substitute the residuals and evaluate, we get a concrete
+            // answer consistent with the partial response
+            let concrete_response = ans
+                .reauthorize(&mapping, &auth, q.clone(), &input.entities)
+                .unwrap();
+            assert!(
+                concrete_response.decision().is_some(),
+                "Substitute all `unknown`s should give a decision"
```
Comment (khieta) [inline cedar-drt/fuzz/fuzz_targets/partial-eval.rs:220]:
```suggestion
                "Substituting all `unknown`s should give a decision"
```

File: cedar-drt/fuzz/fuzz_targets/partial-eval.rs [hunk 2]
```diff
@@ -146,32 +204,32 @@ fuzz_target!(|input: FuzzTargetInput| {
...
+            );
+            let concrete: Response = concrete_response.into();
+            partial_response_correctness(&ans, &concrete);
+
+            // Then we check if we substitute the original policies and evaluate, we get a concrete
+            // ansewr consistent with the partial answer
```
Comment (khieta) [inline cedar-drt/fuzz/fuzz_targets/partial-eval.rs:226]:
```suggestion
            // answer consistent with the partial answer
```

File: cedar-drt/fuzz/fuzz_targets/partial-eval.rs [hunk 3]
```diff
@@ -146,32 +204,32 @@ fuzz_target!(|input: FuzzTargetInput| {
...
+            let substituted_start = substitute_set(&policyset, &mapping);
+            let final_answer = auth.is_authorized(q.clone(), &substituted_start, &input.entities);
+            partial_response_correctness(&ans, &final_answer);
+
+            // Then check that those two concrete respones are equivalent
```
Comment (khieta) [inline cedar-drt/fuzz/fuzz_targets/partial-eval.rs:231]:
```suggestion
            // Then check that those two concrete responses are equivalent
```

File: cedar-drt/fuzz/fuzz_targets/partial-eval.rs [hunk 4]
```diff
@@ -123,6 +128,59 @@ fn responses_equiv(a1: Response, a2: Response) -> bool {
         && (a1.diagnostics.errors.is_empty() == a2.diagnostics.errors.is_empty())
 }
```
Comment (khieta) [inline cedar-drt/fuzz/fuzz_targets/partial-eval.rs:129]: Why not check for exact equality in `responses_equiv`?

**Review by khieta — APPROVED** [review]

File: cedar-drt/fuzz/fuzz_targets/partial-eval.rs [hunk 5]
```diff
@@ -123,6 +128,59 @@ fn responses_equiv(a1: Response, a2: Response) -> bool {
...
+        p.id().clone(),
+        p.annotations_arc().clone(),
+    )
+}
+
+fn get_id(e: &AuthorizationError) -> &PolicyID {
+    match e {
+        AuthorizationError::PolicyEvaluationError { id, .. } => id,
+    }
+}
```
[the remark this answers is not in the document]
Comment (aaronjeline) [reply cedar-drt/fuzz/fuzz_targets/partial-eval.rs:158]: That's not in core, this is using the core api

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**